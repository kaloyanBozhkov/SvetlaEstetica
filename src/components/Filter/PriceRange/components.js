import React, { Fragment } from "react"
import { Box } from '@chakra-ui/core'
import PropTypes from "prop-types"

// *******************************************************
// RAIL
// *******************************************************
const railOuterStyle = {
    position: "absolute",
    width: "100%",
    height: 42,
    transform: "translate(0%, -50%)",
    borderRadius: 7,
    cursor: "pointer"
    // border: '1px solid white',
}

const railInnerStyle = {
    position: "absolute",
    width: "100%",
    height: 2,
    transform: "translate(0%, -50%)",
    borderRadius: 7,
    pointerEvents: "none",
    backgroundColor: "rgb(155,155,155)"
}

export function SliderRail({ getRailProps }) {
    return (
        <Fragment>
            <Box style={railOuterStyle} {...getRailProps()} />
            <Box style={railInnerStyle} />
        </Fragment>
    )
}

SliderRail.propTypes = {
    getRailProps: PropTypes.func.isRequired
}

// *******************************************************
// HANDLE COMPONENT
// *******************************************************
export function Handle({
    domain: [min, max],
    handle: { id, value, percent },
    disabled,
    getHandleProps
}) {
    return (
        <Fragment>
            <Box
                style={{
                    left: `${percent}%`,
                    position: "absolute",
                    transform: "translate(-50%, -50%)",
                    WebkitTapHighlightColor: "rgba(0,0,0,0)",
                    zIndex: 5,
                    width: 28,
                    height: 42,
                    cursor: "pointer",
                    // border: '1px solid white',
                    backgroundColor: "none"
                }}
                {...getHandleProps(id)}
            />
            <Box
                role="slider"
                aria-valuemin={min}
                aria-valuemax={max}
                aria-valuenow={value}
                style={{
                    left: `${percent}%`,
                    position: "absolute",
                    transform: "translate(-50%, -50%)",
                    zIndex: 2,
                    width: 12,
                    height: 12,
                    borderRadius: "50%",
                    boxShadow: "1px 1px 1px 1px rgba(0, 0, 0, 0.3)",
                }}
                bg={disabled ? "#666" : "neutral.200"}
            />
        </Fragment>
    )
}

Handle.propTypes = {
    domain: PropTypes.array.isRequired,
    handle: PropTypes.shape({
        id: PropTypes.string.isRequired,
        value: PropTypes.number.isRequired,
        percent: PropTypes.number.isRequired
    }).isRequired,
    getHandleProps: PropTypes.func.isRequired,
    disabled: PropTypes.bool
}

Handle.defaultProps = {
    disabled: false
}

// *******************************************************
// KEYBOARD HANDLE COMPONENT
// Uses a button to allow keyboard events
// *******************************************************
export function KeyboardHandle({
    domain: [min, max],
    handle: { id, value, percent },
    disabled,
    getHandleProps
}) {
    return (
        <button
            role="slider"
            aria-valuemin={min}
            aria-valuemax={max}
            aria-valuenow={value}
            style={{
                left: `${percent}%`,
                position: "absolute",
                transform: "translate(-50%, -50%)",
                zIndex: 2,
                width: 24,
                height: 24,
                borderRadius: "50%",
                boxShadow: "1px 1px 1px 1px rgba(0, 0, 0, 0.3)",
                backgroundColor: disabled ? "#666" : "#ffc400"
            }}
            {...getHandleProps(id)}
        />
    )
}

KeyboardHandle.propTypes = {
    domain: PropTypes.array.isRequired,
    handle: PropTypes.shape({
        id: PropTypes.string.isRequired,
        value: PropTypes.number.isRequired,
        percent: PropTypes.number.isRequired
    }).isRequired,
    getHandleProps: PropTypes.func.isRequired,
    disabled: PropTypes.bool
}

KeyboardHandle.defaultProps = {
    disabled: false
}

// *******************************************************
// TRACK COMPONENT
// *******************************************************
export function Track({ source, target, getTrackProps, disabled }) {
    return (
        <Box
            style={{
                position: "absolute",
                transform: "translate(0%, -50%)",
                height: 4,
                zIndex: 1,
                borderRadius: 7,
                cursor: "pointer",
                left: `${source.percent}%`,
                width: `${target.percent - source.percent}%`
            }}
            bg={disabled ? "#999" : "neutral.900"}
            {...getTrackProps()}
        />
    )
}

Track.propTypes = {
    source: PropTypes.shape({
        id: PropTypes.string.isRequired,
        value: PropTypes.number.isRequired,
        percent: PropTypes.number.isRequired
    }).isRequired,
    target: PropTypes.shape({
        id: PropTypes.string.isRequired,
        value: PropTypes.number.isRequired,
        percent: PropTypes.number.isRequired
    }).isRequired,
    getTrackProps: PropTypes.func.isRequired,
    disabled: PropTypes.bool
}

Track.defaultProps = {
    disabled: false
}