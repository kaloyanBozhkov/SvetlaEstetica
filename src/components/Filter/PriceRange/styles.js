const styles = {
    sliderRootStyle: {
        position: "relative",
        width: "100%"
    },
    priceRangeWrapper: {
        alignItems: 'center',
        minW: '30rem',
        h: '2.5rem'
    },
    priceLabel: {
        fontWeight: 'bold',
        fontSize: 'lg',
        w: '20rem'
    }
}

export default styles