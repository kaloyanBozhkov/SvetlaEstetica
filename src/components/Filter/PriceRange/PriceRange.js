import React from 'react'
import PropTypes from 'prop-types'
import { Flex, Box, Text } from '@chakra-ui/core'
import { Slider, Rail, Handles, Tracks } from 'react-compound-slider'
import { SliderRail, Handle, Track } from './components'

import styles from './styles'

const PriceRange = ({
    maxPrice = 0,
    minPrice = 0,
    priceRange = [0, 0],
    setPriceRange = (f) => f,
}) => {
    const domain = [minPrice, maxPrice]

    return (
        <Flex {...styles.priceRangeWrapper}>
            <Text {...styles.priceLabel}>{priceRange.toString()}</Text>
            <Slider
                mode={3}
                step={1}
                domain={domain}
                rootStyle={styles.sliderRootStyle}
                onChange={setPriceRange}
                values={priceRange}
            >
                <Rail>
                    {({ getRailProps }) => (
                        <SliderRail getRailProps={getRailProps} />
                    )}
                </Rail>
                <Handles>
                    {({ handles, getHandleProps }) => (
                        <Box>
                            {handles.map((handle) => (
                                <Handle
                                    key={handle.id}
                                    handle={handle}
                                    domain={domain}
                                    getHandleProps={getHandleProps}
                                />
                            ))}
                        </Box>
                    )}
                </Handles>
                <Tracks left={false} right={false}>
                    {({ tracks, getTrackProps }) => (
                        <Box>
                            {tracks.map(({ id, source, target }) => (
                                <Track
                                    key={id}
                                    source={source}
                                    target={target}
                                    getTrackProps={getTrackProps}
                                />
                            ))}
                        </Box>
                    )}
                </Tracks>
            </Slider>
        </Flex>
    )
}

PriceRange.propTypes = {
    maxPrice: PropTypes.number,
    minPrice: PropTypes.number,
    currentMaxPrice: PropTypes.number,
    currentMinPrice: PropTypes.number,
    setMaxPrice: PropTypes.func,
    setMinPrice: PropTypes.func,
}

export default PriceRange
