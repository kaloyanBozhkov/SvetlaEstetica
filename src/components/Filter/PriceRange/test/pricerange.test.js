import React from 'react'
import { shallow } from 'enzyme'

import PriceRange from '../PriceRange'

describe('testing price range', () => {
    it('should render component with default props', () => {
        const wrapper = shallow(<PriceRange />)
        expect(wrapper).toMatchSnapshot()
    })

    it('should trigger onChange fn when interacting with slider', () => {
        const props = {
                maxPrice: 0,
                minPrice: 0,
                priceRange: [0, 0],
                setPriceRange: jest.fn((f) => f),
            },
            mockPriceRange = [20, 80],
            wrapper = shallow(<PriceRange {...props} />),
            slider = wrapper.find('Slider')

        expect(props.setPriceRange).toHaveBeenCalledTimes(0)
        slider.prop('onChange')(mockPriceRange)
        expect(props.setPriceRange).toHaveBeenCalledTimes(1)
        expect(props.setPriceRange.mock.results[0].value).toEqual(mockPriceRange)
    })
})
