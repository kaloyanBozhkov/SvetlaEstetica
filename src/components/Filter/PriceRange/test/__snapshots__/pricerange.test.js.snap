// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing price range should render component with default props 1`] = `
<Flex
  alignItems="center"
  h="2.5rem"
  minW="30rem"
>
  <Text
    fontSize="lg"
    fontWeight="bold"
    w="20rem"
  >
    0,0
  </Text>
  <Slider
    domain={
      Array [
        0,
        0,
      ]
    }
    mode={3}
    onChange={[Function]}
    rootStyle={
      Object {
        "position": "relative",
        "width": "100%",
      }
    }
    step={1}
    values={
      Array [
        0,
        0,
      ]
    }
  >
    <Rail>
      <Component />
    </Rail>
    <Handles>
      <Component />
    </Handles>
    <Tracks
      left={false}
      right={false}
    >
      <Component />
    </Tracks>
  </Slider>
</Flex>
`;
